### PR TITLE
Webi 15881 upgrade the 3rd parties libraries used by the tools to their latest version lfo

### PR DESCRIPTION
**Pull request by leon2Meudon92**
https://jira.castsoftware.com/browse/WEBITOOLS-203

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -75,7 +81,7 @@
             <dependency>
                 <groupId>commons-io</groupId>
                 <artifactId>commons-io</artifactId>
-                <version>2.7</version>
+                <version>2.11.0</version>
```
Comment (Krelix) [inline pom.xml]: Since you're updating the version, can you move them to a property ? This will make it easier to update versions when needed (for this dependency and others below as well)

File: aip-console-jenkins/pom.xml
Comment (Krelix) [inline aip-console-jenkins/pom.xml:1]: Should we also update the jenkins.version that we are using as a "base" version ? The 2.60.3 we released in 2017 so we should probably update it to a more recent version; but not the latest, this property means "this is the minimum version of jenkins you need to use this plugin".

Could you test changing it to 2.303.3, since this is our version as well, and see if everything works fine ?
If it doesn't work, we'll update it later.

File: aip-console-tools-core/pom.xml [hunk 2]
```diff
@@ -50,9 +50,21 @@
...
+                </exclusion>
+                <exclusion>
+                    <groupId>org.jetbrains.kotlin</groupId>
+                    <artifactId>kotlin-stdlib-jdk7</artifactId>
+                </exclusion>
+                <exclusion>
+                    <groupId>org.jetbrains.kotlin</groupId>
+                    <artifactId>kotlin-stdlib-jdk8</artifactId>
+                </exclusion>
+            </exclusions>
```
Comment (Krelix) [inline aip-console-tools-core/pom.xml]: Why did you add those exclusions ? Do they cause issue during the build with enforcer ? I don't see these as dependencies in the okhttp pom; I only see "android" but it is a "provided" scope meaning it shouldn't be added.

Can you try removing them and seeing if you are having any issue ? If you want we can have a call to discuss this.

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java [hunk 3]
```diff
@@ -278,6 +280,26 @@ public int publishToImaging(String applicationName, long sleepDuration, boolean
...
+
+    @Override
+    public String downloadDeliveryReport(String appGuid, String versionGuid, String reportFilename) throws ApplicationServiceException {
+        try {
+            return restApiService.getForEntity("/api/applications/" + appGuid + "/versions/" + versionGuid + "/dmt-report/download", String.class);
+        } catch (ApiCallException e) {
+            throw new ApplicationServiceException("Failed to download the delivery report", e);
+        }
+    }
+
```
Comment (Krelix) [inline aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java:302]: No need to remove this, but I would prefer if you kept the scope of a PR to the specific ticket; this looks like a change from another ticket that should be separated. For this, nothing needs to be changed, just a reminder for other PRs :)

**Review by Krelix — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 1 above)
Comment (leon2Meudon92) [reply pom.xml]: I'll

File: aip-console-jenkins/pom.xml
Comment (leon2Meudon92) [reply aip-console-jenkins/pom.xml:1]: let go for [2.303.3](https://repo.jenkins-ci.org/artifactory/public/org/jenkins-ci/main/jenkins-core/2.303.3/) and see

Didn't work
and here are listed risks https://www.jenkins.io/blog/2018/03/15/jep-200-lts/

I suggest to upgrade later

File: aip-console-tools-core/pom.xml (hunk 2 above)
Comment (leon2Meudon92) [reply aip-console-tools-core/pom.xml]: compile complaint for version clash
And also kotlin not used

File: aip-console-jenkins/pom.xml
Comment (Krelix) [reply aip-console-jenkins/pom.xml:1]: Yeah we'll update later, no big deal.

File: aip-console-tools-core/pom.xml (hunk 2 above)
Comment (Krelix) [reply aip-console-tools-core/pom.xml]: That's weird, but yeah if it causes errors with the enforcer plugin, then keep those around. I'll have a check later.

File: pom.xml (hunk 1 above)
Comment (leon2Meudon92) [reply pom.xml]: done

File: aip-console-jenkins/pom.xml
Comment (leon2Meudon92) [reply aip-console-jenkins/pom.xml:1]: doesn't work by now
Needs slight more works for adoption
https://www.jenkins.io/blog/2018/03/15/jep-200-lts/

File: aip-console-tools-core/pom.xml (hunk 2 above)
Comment (leon2Meudon92) [reply aip-console-tools-core/pom.xml]: sounds well and seems to work with new versions

File: aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java (hunk 3 above)
Comment (leon2Meudon92) [reply aip-console-tools-core/src/main/java/com/castsoftware/aip/console/tools/core/services/ApplicationServiceImpl.java:302]: got it
But was just part of a the refactoring

**Review by Krelix — APPROVED** [review]
